src/Voron/Exceptions/VoronUnrecoverableErrorException.cs

```diff
@@ -6,11 +6,25 @@
 
 using System;
 using System.Runtime.ExceptionServices;
+using Voron.Impl;
 
 namespace Voron.Exceptions
 {
     public class VoronUnrecoverableErrorException : Exception
     {
+        public static void Raise(LowLevelTransaction tx, string message)
+        {
+            try
+            {
+                tx.MarkTransactionAsFailed();
+                throw new VoronUnrecoverableErrorException(message);
```

Comment:
Before we mark the transaction as failed let's add the current state of the transaction to the exception message. I'd like to be able to distinguish if the unrecoverable exception was thrown on a transaction in valid state or already marked as errored (just in case).

Reply:
done